File: backend/models/workflow_validation.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


class ValidationFinding(BaseModel):
    node_id: str | None
    tier: int
    severity: Literal["error", "warning"]
    code: str
    message: str


class WorkflowValidationResult(BaseModel):
    findings: list[ValidationFinding]
    has_errors: bool

File: frontend/src/components/features/workflows/types/workflow-validation.ts
export type ValidationSeverity = "error" | "warning";

export interface ValidationFinding {
  node_id: string | null;
  tier: number;
  severity: ValidationSeverity;
  code: string;
  message: string;
}

export interface WorkflowValidationResult {
  findings: ValidationFinding[];
  has_errors: boolean;
}
